### ISSUE TITLE
Bug: Keeps removing '-' in the middle of a sentence

### ISSUE THREAD

**Issue by jarnnk** [platers/obsidian-linter#287]
## Describe the Bug

I noticed that enumerations in the German language, often abbreviated with '-', are detected as some kind of linting area.
E.g.: "Intermediär- und Terminaloperationen".
Already turned of the "Remove Consectuive List Markers", "Remove Empty List Markers", "No Bare URLs", "Remove Empty Lines Between List Markers and Checklists" and "Remove link spacing" feature. Still no effect.

## How to Reproduce

```
Intermediär- und Terminaloperationen -> Intermediärund Terminaloperationen
```

Please make sure to include the rule(s) that cause the issue if applicable. If you are not sure which rule(s) may cause the issue, please attach the `data.json` file located at `<vault>/.obsidian/plugins/obsidian-linter/`.

## Expected Behavior

Expected output if applicable:

``` 
Intermediär- und Terminaloperationen
```

## Device

- [x] Desktop
- [ ] Not further tested

## Additional Context

[data.json.txt](https://github.com/platers/obsidian-linter/files/9114035/data.json.txt)

**Comment by pjkaufman** [comment]
@jarnnk , the issue is that you have the following rule enabled: Remove Hyphenated Line Breaks.

If a dash comes after a character and is not followed by another character, it will remove it.
For example `car- person` becomes `car person` while `car-person` remains unchanged.

**Comment by jarnnk** [comment]
@pjkaufman Well... thanks your time! This seems to be a case of "my English is to bad for your plugin"...

**Comment by pjkaufman** [comment]
No worries. I am glad I could help.
